logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -0,0 +1,107 @@
+# encoding: utf-8
+require "concurrent"
+require "logstash/event"
+require "logstash/instrument/metric_type"
+
+module LogStash module Instrument
+  # The Metric store the data structure that make sure the data is
+  # saved in a retrievable way, this is a wrapper around multiples ConcurrentHashMap
+  # acting as a tree like structure.
+  class MetricStore
+    class ConcurrentMapExpectedError < Exception; end
+
+    def initialize
+      @store = Concurrent::Map.new
+    end
+
+    # This method use the namespace and key to search the corresponding value of
+    # the hash, if it doesn't exist it will create the appropriate namespaces
+    # path in the hash and return `new_value`
+    #
+    # @param [Array] The path where the values should be located
+    # @param [Object] The default object if the value is not found in the path
+    # @return [Object] Return the new_value of the retrieve object in the tree
+    def fetch_or_store(namespaces, key, default_value = nil)
+      fetch_or_store_namespaces(namespaces).fetch_or_store(key, block_given? ? yield(key) : default_value)
+    end
+    
+    # This method allow to retrieve values for a specific path,
+    #
+    # Also support `*` as a globbing path
```

Comment:
But when I did the code yesterday I found adding the `*` was yielding confusing results and wasnt making sense in my brain. 

With the current features, I don't when this could be useful to support a `*` ? Whats your opinion on that?